book/content/code_styling/Code_Styling.md

```diff
@@ -0,0 +1,279 @@
+It also means you keep to the 80 character line length suggestion.
+Also, in RStudio you can fold away the comment using the arrow next to the 
+line number of the if statement.
+}
+```
+
+Or commenting out individual lines:
+
+```{r}
+#This is also a very long comment
+#covering many lines.
+```
+Your IDE will probably have a keyboard shortcut for commenting out blocks though.
+
+## Indentation
+The R style guide suggests that lines should be separated:
+```
+by
+  two spaces
+```
+And not
+```
+ a mixture 
+   of 
+   	tabs 
+   	  and 	spaces.
+```
+
+Obviously, sometimes the arguments of a function can far expand 80 characters.
+In this case, it is recommended that the second line be indented to the start of the arguments:
+
+```
+my_variable <- a_really_long_function(data = "2019-05-17_Long_File_Name_2",
+                                      header = TRUE, verbose = TRUE)
+
+```
+
+These are of course just guidelines, and you should choose elements that suit your own coding style.
+However, and again, it is important to ensure that you are consistent when collaborating, and can agree on a common style.
+It could be useful to create a readme file describing your coding style so collaborators or contributors can follow your lead.
+
+
+### ...end. ...end.  ...or end.\\n
+
+If you are sharing text files or working collaboratively on manuals or documents, then there is a lot of controversy surrounding whether to use one or two spaces after a period.
+When using Markdown, it can actually be clearer to include a new line after every sentence.^[This chapter (and most, if not all, of this book) has a new line after every sentence]
+This makes the raw text easier to read, and solves the spacing issue.
+
+![We don't have spears though: http://xkcd.com/1285/](xkcd1285.png)
+
+# Tools
+As mentioned earlier, there are some automatic tools that you can use to lint your code to existing guidelines.
+These range from plugins for IDEs, packages that 'spell check' your style, and scripts that automatically lint for you.
+
+## lintr
+lintr is an R package that spell checks your code using a variety of style guidelines.  It can be installed from CRAN.
+The function `lint` takes a filename as an argument, and a list of 'linters' that it should check your code against.
+These range from whitespace conventions to checking that curly braces do not have their own lines.
+
+![](lintr_1.png)
+![](lintr_2.png)
+
+Lintr provides a list of markers with recommendations for changing the formatting of your code line-by-line, meaning it is best used early and often in your project.
+
+## Autopep8
+Autopep8 is a Python module that can be run from the terminal and automatically formats a file to PEP8 guidelines.  It is available on [pypy](https://pypi.org/project/autopep8/), and can be installed using pip.
+![](console.png)
+![Before pep8](pre_pep.png)
+![After pep8](pep.png)
+
+To some extent, the module can also be used on R scripts!
+![](pep8_R.png)
+
```

Comment:
```suggestion

## Black
[Black](https://black.readthedocs.io/en/stable/) is an auto-formatter for Python.
This means that it will automatically change your code to adhere to certain guidelines, like line-length.
It is also consistent, so that the code that you and your collaborators work on, will look the same once black formats it.
It does not change what the code does.
This can reduce the time spent making the above changes to the code.

```